packages/concerto-core/lib/introspect/stringvalidator.js

```diff
@@ -71,7 +71,7 @@ class StringValidator extends Validator{
                 this.regex = new CustomRegExp(validator.pattern, validator.flags);
             }
             catch(exception) {
-                this.reportError(field.getName(), exception.message, ErrorCodes.REGEX_VALIDATOR_EXCEPTION);
+                this.reportError(field.getName(), exception.message, ErrorCodes.REGEX_VALIDATOR_EXCEPTION.code, ErrorCodes.REGEX_VALIDATOR_EXCEPTION.status);
```

Comment:
What is the benefit here of splitting the code and status into two parameters? Would one suffice?

Reply:
We can surely reduce it to the {code,status} object that we want and pass it directly, instead of spreading it. This approach was made keeping in mind that there wasn't any grouping of errors defined explicitly. So the user can perhaps use a separate error code and status to define in an exception. I will reduce it to a single object.